packages/adapter/addon/rest.ts

```diff
@@ -902,11 +938,11 @@ class RESTAdapter extends Adapter.extend(BuildURLMixin) {
     @param  {Object} requestData - the original request information
     @return {Object | AdapterError} response
   */
-  handleResponse(status, headers, payload, requestData) {
+  handleResponse(status: number, headers, payload: Payload, requestData: RequestData) {
     if (this.isSuccess(status, headers, payload)) {
       return payload;
     } else if (this.isInvalid(status, headers, payload)) {
-      return new InvalidError(payload.errors);
+      return new InvalidError((payload as Dict<unknown>).errors);
```

Comment:
Why do we need a cast here

Reply:
Payload can be `string | Dict<unknown> | undefined`